dialect_cockroach_test.go

```diff
@@ -0,0 +1,39 @@
+package pop
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Cockroach_URL_Raw(t *testing.T) {
+	r := require.New(t)
+	cd := &ConnectionDetails{
+		Dialect: "cockroach",
+		URL:     "scheme://user:pass@host:port/database?option1=value1",
+	}
+	err := cd.Finalize()
+	r.NoError(err)
+	m := &cockroach{ConnectionDetails: cd}
+	r.Equal("scheme://user:pass@host:port/database?option1=value1", m.URL())
```

Comment:
I believe we always should respect user's configuration even if the value is not correct.
In this *UNIT* test, it just checks result of functions `URL()` and `urlWithoutDb()` and not the validity of the value.
In the case of `URL()`, I want to check it was modified or not. If a user gives a  wrong URL value, it is better to make an error rather than try to correct it automatically. It will more helpful to prevent hidden problems.
For `urlWithoutDb()`, I just preserved the method which is building a new string from connection details, but I think it should be changed as just remove the database part from original URL.